NanoRange failed to build due to errors C2677 C3536 C2893 C2062 C2607 C2672 C2676 with MSVC on windows

Environment:
Windows Server 2016 + VS2017 + NanoRange master branch latest srouce code.

NanoRange failed to build due to errors C2677 C3536 C2893 C2062 C2607 C2672 C2676 with MSVC on windows. It can be reproduced on master revision 391cc95 . Could you help have a look about this issue? Thanks in advance! 

Steps to reproduce the behavior:
1.Open VS2017 x86 Native tools command tool
2.git clone https://github.com/tcbrindle/nanorange d:\NanoRange\src
3.cd D:\NanoRange
4.mkdir build_x86 && pushd build_x86
5.cmake -G "Visual Studio 15 2017" -DCMAKE_SYSTEM_VERSION=10.0.17134.0 -DCMAKE_CXX_STANDARD=17 ..\src\
6.msbuild /m /p:Configuration=Release;Platform=Win32 nanorange.sln /t:Rebuild

[log_x86_build.log](https://github.com/tcbrindle/NanoRange/files/3491364/log_x86_build.log)

Actual result:
D:\NanoRange\src\test\view\filter_view.cpp(45): error C2677: binary '|': no global operator found which takes type 
D:\NanoRange\src\test\view\filter_view.cpp(48): error C3536: 'rng': cannot be used before it is initialized [D:\NanoRange\build_x86\test\test_nanorange.vcxproj]
D:\NanoRange\src\test\view\filter_view.cpp(48): error C2893: Failed to specialize function template 'unknown-type nano::ranges::detail::begin_::fn::operator ()(T &&) noexcept(<expr>) const' [D:\NanoRange\build_x86\test\test_nanorange.vcxproj]
D:\NanoRange\src\test\view\filter_view.cpp(48): error C2062: type 'unknown-type' unexpected [D:\NanoRange\build_x86\test\test_nanorange.vcxproj]
D:\NanoRange\src\test\view\filter_view.cpp(49): error C2607: static assertion failed [D:\NanoRange\build_x86\test\test_nanorange.vcxproj]
D:\NanoRange\src\test\view\filter_view.cpp(141): error C2672: 'operator __surrogate_func': no matching overloaded function found [D:\NanoRange\build_x86\test\test_nanorange.vcxproj]
D:\NanoRange\src\test\view\filter_view.cpp(135): error C2676: binary '|': 'nano::ranges::iota_view<_Ty,nano::ranges::unreachable_sentinel_t>' does not define this operator or a conversion to a type acceptable to the predefined operator [D:\NanoRange\build_x86\test\test_nanorange.vcxproj]

--- a/include/nanorange/detail/macros.hpp
+++ b/include/nanorange/detail/macros.hpp
@@ -51,7 +51,7 @@ inline namespace ranges                                                        \
 #define NANO_END_NAMESPACE_STD }
 #endif
 
-#if defined(_MSC_VER) && _MSC_VER < 1921
+#if defined(_MSC_VER) && _MSC_VER < 1923
 #define NANO_MSVC_LAMBDA_PIPE_WORKAROUND 1
 #endif
 

